dom/src/main/scala/org/scalajs/dom/ResizeObserver.scala

```diff
@@ -0,0 +1,24 @@
+package org.scalajs.dom
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+/** The ResizeObserver constructor creates a new ResizeObserver object, which can be used to report changes to the
+  * content or border box of an Element or the bounding box of an SVGElement
+  *
+  * @param callback
+  *   The function called whenever an observed resize occurs.
+  */
+@js.native
+@JSGlobal
+class ResizeObserver(callback: js.Function2[js.Array[ResizeObserverEntry], ResizeObserver, _]) extends js.Object {
+
+  /** Starts observing the specified Element or SVGElement. */
+  def observe(target: Element, options: js.UndefOr[ResizeObserverOptions] = js.native): Unit = js.native
```

Comment:
Ah, sorry, one more thing. Let's write this as overloads, since this works better for type inference esp. on Scala 3. I recently learned this technique in https://github.com/scala-js/scala-js-dom/pull/691#discussion_r844629202.

```suggestion
  def observe(target: Element): Unit = js.native
  def observe(target: Element, options: ResizeObserverOptions): Unit = js.native
```

Reply:
Oh interesting, thats good to know!